### ISSUE TITLE
Prevent multiple presenters clobbering eachother's results

### ISSUE THREAD

**Issue by nathansherburn** [SLC3/MARS#110]
Jon and I (and Faulksy?) discussed the idea of showing "Lecturer A" a "This poll is in use" page until "Lecturer B's" result view has been closed.

The risk is Lecturer B sitting at his desk keeping the result page open too long and preventing Lecturer A from running her polls. Perhaps we could include an indicator on Lecturer B's screen to show that Lecturer A is waiting to start the poll.

We could also make it so that Lecturer A is able to press a button to kick Lecturer B out forcefully. We'd have to provide Lecturer B a notice like "Lecturer A has just forcefully ejected you from this poll in order to gain access".

As funny as the second option sounds, it may be the way to go given that you don't want to be stuck in a lecturer ready to start a poll and have to awkwardly wait for an indefinite amount of time before your colleague sees that you need access.

**Comment by aj1331** [comment]
We could look at a timeout option before allowing one lecturer to kick another out. Something like if the time since the last response was received is greater than 5min, allow Lecturer A to kick Lecturer B out.

**Comment by nathansherburn** [comment]
Yeah I like that idea.

Perhaps just before getting booted, Lecturer B should get a notification saying "It looks like one of your collaborators is trying to run this poll - Reject - Allow".

We could give this dialog a timer that automatically chooses "Allow" after 15 sec

**Comment by thanatos6666** [comment]
Perhaps the easier alternative is to just allow the two lecturers to open
two simultaneous polls, but the + symbol pops up and students switch
between. Not sure what happens if lecturers open the SAME poll (new
instance of same poll?)

**Comment by nathansherburn** [comment]
I think simultaneous polls work at the moment. It's just the same poll situation that's a bit weird because if A stumbles into B's poll, as soon as A exits the poll, the poll will end (disrupting B's class).

**Comment by thanatos6666** [comment]
So are you saying saying that both A and B opening the same poll both get
the same instance - and in this case, A/B closing causes B/A to also close?

OR

Both A and B will get different instances, but A/B closing still causes B/A
to also close?

**Comment by nathansherburn** [comment]
They both get the same instance of the poll but that means as soon as either one of them leaves, the polls over.

**Comment by aj1331** [comment]
How does the student know which lecturer's poll they're responding to? The screen will be identical for them which will make it look like the + button isn't working.

**Comment by nathansherburn** [comment]
When A and B open the same poll, there's only one open result so the students don't get a pink "next poll" button.

When A and B open different polls, there are two results and students get the pink "next poll" button.

**Comment by thanatos6666** [comment]
Right, then considering Andrew's question, can we just have a popup:

"This poll appears currently to be in use by a colleague. Please be aware
that if you exit this poll after joining it, you will close the poll for
everyone. Do you still wish to join? Yes/No"

**Comment by nathansherburn** [comment]
Yeah that works. I just tested and found that it actually _pauses_ the poll (not closes it) for the lecturer who doesn't leave.
